### PR TITLE
Feature/new approach game

### PR DESCRIPTION
**Pull request by JorxSS**
What's new:

* New Light Ball cast
* Camera on rails
* New game/level approach
* Scene to playground test fast things

In order to be able to work in this branch you will need to have [this](https://github.com/OnionGalaxy/LittleOrionEngine/pull/205)

KNOWN BUG:
Mesh collider doesn't respect the rotation of the mesh.

### CODE REVIEW

File: Scripts/src/Script/CameraController.cpp [hunk 1]
```diff
@@ -144,157 +134,60 @@ void CameraController::ActivePlayer()
...
-	}
+	focus_progress = current_time / center_time_position;
+	float3 new_camera_position = Quat::SlerpVector(previous_position,position_to_focus, focus_progress);
 	owner->transform.SetGlobalMatrixTranslation(new_camera_position);
-	is_focusing = owner->transform.GetGlobalTranslation().x == position_to_focus.x;
+	is_focusing = focus_progress < 1;
 }
 
-void CameraController::SinglePlayerCamera()
+void CameraController::RotateFocus(float3& position_to_rotate)
```
Comment (Dartemiss) [inline Scripts/src/Script/CameraController.cpp:143]: const float3& position_to_rotate, mi pana

File: Scripts/src/Script/CameraController.cpp [hunk 2]
```diff
@@ -144,157 +134,60 @@ void CameraController::ActivePlayer()
 
 void CameraController::Focus(float3& position_to_focus)
```
Comment (Dartemiss) [inline Scripts/src/Script/CameraController.cpp:135]: const float3& position_to_rotate, mi pana

File: Scripts/src/Script/EnemyManager.cpp [hunk 3]
```diff
@@ -312,3 +312,14 @@ bool EnemyManager::CheckSpawnAvailability(float3& spawn_position)
 
 	return true;
 }
+
+void EnemyManager::KillAllTheEnemies()
+{
+	for(auto& enemy : enemies)
+	{
+		enemy->is_alive = false;
+		enemy->collider->active_physics = false;
```
Comment (Dartemiss) [inline Scripts/src/Script/EnemyManager.cpp:321]: Is this needed?

File: Scripts/src/Script/CameraController.cpp [hunk 4]
```diff
@@ -142,159 +132,62 @@ void CameraController::ActivePlayer()
...
-
+	float3 desired_direction = position_to_look_at - owner->transform.GetGlobalTranslation();
+	Quat desired_rotation = Quat::LookAt(float3::unitZ, desired_direction.Normalized(), float3::unitY, float3::unitY);
+	Quat new_rotation = Quat::Slerp(owner->transform.GetRotation(), desired_rotation, App->time->delta_time * smoothing_time);
+	owner->transform.SetRotation(new_rotation);
 }
 
-void CameraController::MultiplayerCamera()
+float3 CameraController::GetPositionToProgress()
 {
```
Comment (marcorod94) [inline Scripts/src/Script/CameraController.cpp:152]: This shoudn't be a const

File: Scripts/src/Script/PlayerController.cpp [hunk 5]
```diff
@@ -306,13 +320,13 @@ void PlayerController::TakeDamage(float damage)
 	UpdateHealthBar();
 }
 
-void PlayerController::Heal(float healed_value)
+void PlayerController::Heal(const float healed_value)
```
Comment (marcorod94) [inline Scripts/src/Script/PlayerController.cpp:323]: it's necessary to make a float const

File: Scripts/src/Script/PlayerController.cpp [hunk 6]
```diff
@@ -306,13 +320,13 @@ void PlayerController::TakeDamage(float damage)
...
 {
 	health_points = math::Min(health_points + healed_value, total_health);
 	UpdateHealthBar();
 }
 
-void PlayerController::FillSpecialBar(float special_value)
+void PlayerController::FillSpecialBar(const float special_value)
```
Comment (marcorod94) [inline Scripts/src/Script/PlayerController.cpp:329]: it's necessary to make a float const

File: Scripts/src/Script/PlayerController.cpp [hunk 7]
```diff
@@ -306,13 +320,13 @@ void PlayerController::TakeDamage(float damage)
 	UpdateHealthBar();
```
Comment (marcorod94) [inline Scripts/src/Script/PlayerController.cpp:320]: it's necessary to make a float const

**Pull request merged**